src/workerd/api/queue.c++

```diff
@@ -179,19 +187,23 @@ kj::Promise<void> WorkerQueue::send(jsg::Lock& js,
 
   JSG_REQUIRE(!body.isUndefined(), TypeError, "Message body cannot be undefined");
 
+  auto headers = kj::HttpHeaders(context.getHeaderTable());
+  headers.set(kj::HttpHeaderId::CONTENT_TYPE, MimeType::OCTET_STREAM.toString());
+
   kj::Maybe<kj::StringPtr> contentType;
   KJ_IF_SOME(opts, options) {
     KJ_IF_SOME(type, opts.contentType) {
-      contentType = validateContentType(type);
+      auto validatedType = validateContentType(type);
+      headers.add(HDR_MSG_FORMAT, validatedType);
+      contentType = validatedType;
+    }
+    KJ_IF_SOME(secs, opts.delaySeconds) {
+      headers.add(HDR_MSG_DELAY, kj::str(secs));
```

Comment:
Since `delaySeconds` is defined as an `int`, what happens in a negative value is provided?

Reply:
We agreed on doing that validation on the server side. I had it as a `uint` before and the error that was thrown was not super descriptive and user-friendly, that's why I changed it to an `int`.